File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
from heapq import heappop, heappush

def solution(scoville, K):
    answer = 0
    count = 0
    heap = []
    for i in range(len(scoville)):
        heappush(heap, scoville[i])
        
    while heap[0]<K:
        if len(heap)<2:
            return -1
        num1 = heappop(heap)
        num2 = heappop(heap)
        answer = num1+2*num2
        count += 1
        heappush(heap, answer)
    
    return count

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        for (int i: scoville) {
            heap.add(i);
        }
        
        while (heap.peek() < K) {
            if (heap.size()<2) return -1;
            int num1 = heap.poll();
            int num2 = heap.poll();
            int temp = num1 + 2 * num2;
            answer++;
            heap.add(temp);
        }
        
        return answer;
    }
}